src/main/java/appeng/worldgen/QuartzWorldGen.java

```diff
@@ -67,26 +72,31 @@ public void generate( final Random r, final int chunkX, final int chunkZ, final
 			seaLevel = w.getHeight( x, z );
 		}
 
-		if( this.oreNormal == null || this.oreCharged == null )
-		{
-			return;
-		}
-
 		final double oreDepthMultiplier = AEConfig.instance().getQuartzOresClusterAmount() * seaLevel / 64;
 		final int scale = (int) Math.round( r.nextGaussian() * Math.sqrt( oreDepthMultiplier ) + oreDepthMultiplier );
 
-		for( int x = 0; x < ( r.nextBoolean() ? scale * 2 : scale ) / 2; ++x )
+		for( int cnt = 0; cnt < ( r.nextBoolean() ? scale * 2 : scale ) / 2; ++cnt )
 		{
-			final boolean isCharged = r.nextFloat() > AEConfig.instance().getSpawnChargedChance();
-			final WorldGenMinable whichOre = isCharged ? this.oreCharged : this.oreNormal;
+			boolean isCharged = false;
+
+			if( this.oreCharged != null )
+			{
+				isCharged = r.nextFloat() > AEConfig.instance().getSpawnChargedChance();
+			}
 
-			if( WorldGenRegistry.INSTANCE.isWorldGenEnabled( isCharged ? WorldGenType.CHARGED_CERTUS_QUARTZ : WorldGenType.CERTUS_QUARTZ, w ) )
+			final WorldGenMinable whichOre = isCharged ? this.oreCharged : this.oreNormal;
+			if( whichOre != null && ShouldGenerate( isCharged, w ) )
 			{
-				final int cx = chunkX * 16 + r.nextInt( 22 );
+				final int cx = chunkX * 16 + r.nextInt( 16 );
 				final int cy = r.nextInt( 40 * seaLevel / 64 ) + r.nextInt( 22 * seaLevel / 64 ) + 12 * seaLevel / 64;
-				final int cz = chunkZ * 16 + r.nextInt( 22 );
+				final int cz = chunkZ * 16 + r.nextInt( 16 );
 				whichOre.generate( w, r, new BlockPos( cx, cy, cz ) );
 			}
 		}
 	}
+
+	private static boolean ShouldGenerate( final boolean isCharged, final World w )
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00100)